testsuite/forge/src/success_criteria.rs

```diff
@@ -30,57 +42,96 @@ pub struct SuccessCriteria {
 }
 
 impl SuccessCriteria {
-    pub fn new(
-        tps: usize,
-        max_latency_ms: usize,
-        check_no_restarts: bool,
-        wait_for_all_nodes_to_catchup: Option<Duration>,
-        system_metrics_threshold: Option<SystemMetricsThreshold>,
-        chain_progress_check: Option<StateProgressThreshold>,
-    ) -> Self {
+    pub fn new(tps: usize) -> Self {
```

Comment:
Since you have refactored this into builder patter now, it would be good to have tps check as well part of that pattern instead of passing this in the constructor for consistency.